### PR TITLE
Update Monk to latest version

### PR DESCRIPTION
**Pull request by thedebian**
### Summary

This PR updates Monk to the latest version, removing a lot of dependencies and vulnerabilities.

### Testing

* Spawn redis and mongo on your cloudup-vm
* `npm install`
* `make test`
* ✅ Tests should pass

Also test alongside Automattic/thumbs-up#25

### CODE REVIEW

**Review by jgcaruso — CHANGES_REQUESTED** [review]
I'm getting a test failure, I wonder if something changed in mongodb versions? I made sure that my docker container was running with redis and mongo though, so it should be the same version you were running.

image

**Comment by thedebian** [comment]
> I'm getting a test failure, I wonder if something changed in mongodb versions? I made sure that my docker container was running with redis and mongo though, so it should be the same version you were running.
> 
> image

That's weird, according to [the documentation](https://www.mongodb.com/docs/manual/reference/operator/update/set/#behavior) "Starting in MongoDB 5.0, [mongod](https://www.mongodb.com/docs/manual/reference/program/mongod/#mongodb-binary-bin.mongod) no longer raises an error when you use an update operator like [$set](https://www.mongodb.com/docs/manual/reference/operator/update/set/#mongodb-update-up.-set) with an empty operand expression ( { } ). An empty update results in no changes and no [oplog](https://www.mongodb.com/docs/manual/reference/glossary/#std-term-oplog) entry is created (meaning that the operation is a no-op)." (and it seems I didn't change this update anyway).

I'll check, either to try the new param we're sending on connection create (the one I don't remember the name), or to update the test to set something, and check/update if there are empty $set in code using mydb-driver

File: index.js [hunk 1]
```diff
@@ -98,6 +112,37 @@ function Collection(manager, name){
 
 Collection.prototype.__proto__ = MonkCollection.prototype;
 
+Collection.prototype.ensureEventEmitterExists = function() {
```
Comment (pgk) [inline index.js:115]: Is this somehting we added for compatibility reasons? If not, it feels like it's use could be more of a lazyload getter. That returns the emitter.

File: index.js [hunk 2]
```diff
@@ -98,6 +112,37 @@ function Collection(manager, name){
...
+  if ( ! this._eventEmitter ) {
+    this._eventEmitter = new EventEmitter();
+  }
+};
+
+Collection.prototype.on = function() {
+  this.ensureEventEmitterExists();
+  return this._eventEmitter.on( ...arguments );
```
Comment (pgk) [inline index.js:123]: If the above comment is true and we choose to make emiter a lazy getter, this and the following can be replaced with 
```js
  this.getEventEmitter().on( ...arguments );
```

**Review by pgk — CHANGES_REQUESTED** [review]
Test passed for me, though the process "hanged". Some feedback on eventEmitter

**Review by pgk — APPROVED** [review]
does the thing!

**Review by jgcaruso — APPROVED** [review]
`make test` passes after upgrading local mongodb to v6.

**Pull request merged**